de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateBL.java

```diff
@@ -1006,46 +1007,25 @@ else if (diffPercent.signum() < 0 && diffPercent.abs().compareTo(conditions.getM
 		dataEntry.setActualQtyDiffPercentEff(effectiveDiffPercent);
 		dataEntry.setC_Currency_ID(currencyId);
 		dataEntry.setFlatrateAmtCorr(amtCorrection);
-
-		// System.out.println("--updateEntry--");
-		// System.out.println("AuxEntry=" + auxEntry);
-		// System.out.println("PricePerUnit=" + pricePerUnit);
-		// System.out.println("ActualQtyDiffAbs=" + actualQtyDiffAbs);
-		// System.out.println("ActualQtyPerUnit=" + actualQtyPerUnit);
-		// System.out.println("Qty_Reported=" + dataEntry.getQty_Reported());
-		// System.out.println("FlatrateAmt=" + flatrateAmt);
-		// System.out.println("FlatrateAmtCorr=" + amtCorrection);
 	}
 
 	@Override
-	public void extendContract(
-			final I_C_Flatrate_Term currentTerm,
-			final boolean forceExtend,
-			final Boolean forceComplete,
-			final Timestamp nextTermStartDate,
-			final I_C_OrderLine ol)
+	public void extendContract(final @NonNull ContractExtendingRequest context)
 	{
+		Services.get(ITrxManager.class).run(ITrx.TRXNAME_ThreadInherited, (TrxRunnable)localTrxName -> {
```

Comment:
i think we don't need the cast to `TrxRunnable`

Reply:
Yes, is added by default by eclipse, due to a setting